mlrun/projects/project.py

```diff
@@ -133,12 +134,19 @@ def new_project(
     :param description:  text describing the project
     :param subpath:      project subpath (relative to the context dir)
     :param save:         whether to save the created project in the DB
+    :param overwrite:    overwrite project if project with name exists
 
     :returns: project object
     """
     context = context or "./"
     name = _add_username_to_project_name_if_needed(name, user_project)
 
+    if overwrite:
+        logger.info(f"Deleting project {name} from MLRun DB due to overwrite")
+        _delete_project_from_db(
+            name, secrets, mlrun.api.schemas.DeletionStrategy.cascade
+        )
```

Comment:
Please move this block to the end of the function, right before you actually `save()` the new project. The way it's written now you can be stuck in a situation where your old project was already deleted, but the new project was not created because the load from template failed or some other failure. We want to minimize the possibility of such a thing happening.